### DISCUSSION TITLE
Objects in custom subs spawn in shifted on purchase or gradually shift as campaign progresses.

### DISCUSSION THREAD

**Discussion by inmanw** [FakeFishGames/Barotrauma#15132] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

In 2 of my custom subs using thalamus and alien items certain items,alien generator (new), guardian pod, incubation chamber, alien toxin generator, gravity sphere, alien storage vat, fleshgun ammo sack, thalamus brain, and cell spawn organ move slightly every after every mission or spawn in shifted. The attached images show a sub purchased in the Great sea in a solo campaign with the guardian pods shifted. The other image shows the guardian pods in their proper place over the diving suits in a multiplayer mission.
![PXL_20241112_191940692](https://github.com/user-attachments/assets/f67451cd-0b5f-4892-bef0-a744d437138c)
![PXL_20241112_192919153 MP](https://github.com/user-attachments/assets/0e3a4050-cdff-47b1-ba64-f3a3db04c3bf)
Undocking and re docking causes the items to shift as well as quiting and reloading the game.

![1731469960028242796823652916913](https://github.com/user-attachments/assets/a9f375f7-2480-4e51-93f3-851feb8693da)

### Reproduction steps

Campaign solo
1 purchase sub 
2 reload/undock
3 items are shifted

Missions/ campaign multiplayer
1 select sub
2 load in
3 No items are shifted
4 undock redock
5 Items are shifted

Sub editor 
1 open vanilla sub (typhoon 2)
2 add any of the listed items
3 save as new local package
4 open multiplayer campaign with edited sub
5 undock/redock
6 items shifted

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Both

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by inmanw** [comment]
Follow up after testing a few things. 
Using a save de compressor I copied the file from local mods into the decompressed save file copying over the same file and loaded the game. This fixed the shifted items but at the loss of all transfered items.
Creating a new local mod package under a different name but identical in every other way and then purchasing the copy in game fixes the shift at the cost of a new sub.
I am unsure if this will fix the gradual shifting of items after missions. 
Closing the game and reloading causes items to shift again. 
Uninstalling all mods and starting a fresh campaign will still cause items to shift when undocking.

**Comment by Regalis11** [comment]
This has been addressed in our private repo now, fix coming up in the next unstable build (and the next public release)!
